core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -135,6 +141,11 @@ public static StructLikeSet toEqualitySet(
     }
   }
 
+  @VisibleForTesting
+  static void setDeletePosThreadPool(ExecutorService threadPool) {
```

Comment:
This is okay, but I think it is better to be able to set the threadpool when running an operation that uses one instead of modifying shared state.

How about changing `toPositionIndex` like this?

```java
  public static <T extends StructLike> PositionDeleteIndex toPositionIndex(
      CharSequence dataLocation, List<CloseableIterable<T>> deleteFiles) {
    return toPositionIndex(dataLocation, deleteFiles, ThreadPools.getDeleteWorkerPool());
  }

  public static <T extends StructLike> PositionDeleteIndex toPositionIndex(
      CharSequence dataLocation, List<CloseableIterable<T>> deleteFiles, ExecutorService threadPool) {
    ...
    if (positions.size() > 1 && threadPool != null) {
      return toPositionIndex(new ParallelIterable<>(positions, deletePosThreadPool));
    } else {
      return toPositionIndex(CloseableIterable.concat(positions)));
    }
  }
```